# ######################
# ### Python pip install
# ######################
# python3-pip, python3-psutil, (python3-pyinotify): Installiert in packages_workstation-basic.yml

# - name: Install packages - pip (Python) (env_user, all, not EnveavourOS)
#   become: true
#   become_user: "{{ env_user }}"
#   ansible.builtin.pip:
#     state: present
#     name:
#       - deepl                     # Ulauncher Extension Requirement: DeepL Translator
#       - faker                     # Ulauncher Extension Requirement: Faker
#       - lorem                     # Ulauncher Extension Requirement: Lipsum
#       - parsedatetime             # Ulauncher Extension Requirement: Calculate Anything
#       - pint                      # Ulauncher Extension Requirement: Calculate Anything
#       - pydbus                    # Ulauncher Extension Requirement: Bluetooth
#       - simpleeval                # Ulauncher Extension Requirement: Calculate Anything
#       - wrapt_timeout_decorator   # Ulauncher Extension Requirement: Bluetooth
#     extra_args: --user
#   when:
#     - ansible_distribution in ["Archlinux", "Ubuntu"]
#     - env_lsb_id != "EndeavourOS"

# !!!
# - pyenv:
#   - https://technicalnavigator.in/how-to-fix-error-externally-managed-environment-in-python-kali-linux/
#
# - allgemein:
# - https://pythonspeed.com/articles/externally-managed-environment-pep-668/
# - https://stackoverflow.com/questions/75616371/sudo-pip-install-cryptography-error-externally-managed-environment
# - https://itsfoss.com/externally-managed-environment/#option-2-use-python-virtual-environment
# - https://stackoverflow.com/questions/75608323/how-do-i-solve-error-externally-managed-environment-everytime-i-use-pip3
# - https://stackoverflow.com/questions/75602063/pip-install-r-requirements-txt-is-failing-this-environment-is-externally-manag/75696359#75696359
# - https://peps.python.org/pep-0668/#mark-the-installation-as-externally-managed
# !!!

- name: Python - pip - Info zu möglicher Fehlermeldung (env_user, all)
  become: true
  become_user: "{{ env_user }}"
  ansible.builtin.debug:
    msg: "Wenn es im Folgeschritt zu Fehlermelung kommt: source .zshrc/.bashrc oder Terminal neu starten und nochmal Playbook starten."

- name: Python - pip - upgrade pip (env_user, all)
  become: true
  become_user: "{{ env_user }}"
  ansible.builtin.shell: pip install --upgrade pip
  args:
    creates: "/home/{{ env_user }}/.ansible_PythonPip"

- name: Python - pip - Installation verschiedene Pakete für Ulauncher extensions (env_user, all)
  become: true
  become_user: "{{ env_user }}"
  ansible.builtin.shell: pip install --user bs4 deepl faker htmlentities parsedatetime pint pydbus pytz simpleeval wrapt_timeout_decorator
  args:
    creates: "/home/{{ env_user }}/.ansible_PythonPip"

- name: Python - pip - set flag file for pip installs (env_user, all)
  become: true
  become_user: "{{ env_user }}"
  ansible.builtin.shell: "touch /home/{{ env_user }}/.ansible_PythonPip"
  args:
    creates: "/home/{{ env_user }}/.ansible_PythonPip"
